### PR TITLE
docs: port push notification guide

### PR DESCRIPTION
**Pull request by imtoori**

### CODE REVIEW

File: docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx [hunk 1]
```diff
@@ -13,24 +13,223 @@ of new messages and old notifications sometimes need to be updated silently as w
...
+```dart
+firebaseMessaging.onTokenRefresh.listen((token) { 
+      client.addDevice(token, PushProvider.firebase); 
+}); 
+```
+
+### Possible issues
+
+
+We only send push notifications, when the user doesn't have any active websocket connection (which is established when you call client.connectUser). If you set the [onBackgroundEventReceived](https://pub.dev/documentation/stream_chat_flutter/latest/stream_chat_flutter/StreamChat/onBackgroundEventReceived.html) property of the StreamChat widget, when your app goes to background, your device will keep the ws connection alive for 1 minute, and so within this period, you won't receive any push notification.
```
Comment (deven98) [inline docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx:96]: Unnecessary comma

Comment (deven98) [inline docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx:96]: Unnecessary comma and maybe add a possible highlight

```suggestion
We only send push notifications when the user doesn't have any active websocket connection (which is established when you call `client.connectUser`). If you set the [onBackgroundEventReceived](https://pub.dev/documentation/stream_chat_flutter/latest/stream_chat_flutter/StreamChat/onBackgroundEventReceived.html) property of the StreamChat widget, when your app goes to background, your device will keep the ws connection alive for 1 minute, and so within this period, you won't receive any push notification.
```

File: docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx [hunk 2]
```diff
@@ -13,24 +13,223 @@ of new messages and old notifications sometimes need to be updated silently as w
...
+        iOS: IOSNotificationDetails(), 
+      ), 
+    ); 
+  }, 
+  child: .... 
+); 
+... 
+```
+
+As you can see we §generate a local notification whenever a message.new or notification.message_new event is received.
```
Comment (deven98) [inline docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx:188]: Is the '§' doing something or symbolic of something? Remove if a typo.

File: docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx [hunk 3]
```diff
@@ -13,24 +13,222 @@ of new messages and old notifications sometimes need to be updated silently as w
...
+### Possible issues
+
+
+We only send push notifications when the user doesn't have any active websocket connection (which is established when you call `client.connectUser`). If you set the [onBackgroundEventReceived](https://pub.dev/documentation/stream_chat_flutter/latest/stream_chat_flutter/StreamChat/onBackgroundEventReceived.html) property of the StreamChat widget, when your app goes to background, your device will keep the ws connection alive for 1 minute, and so within this period, you won't receive any push notification.
+
+Make sure to read the [general push docs](https://getstream.io/chat/docs/flutter-dart/push_introduction/?language=dart) in order to avoid known gotchas that may make your relationship with notifications go bad 😢
+
+### Testing if Push Notifications are Setup Correctly
+
+If you're not sure if you've set up push notifications correctly (eg you don't always receive them, they work unreliably), you can follow these steps to make sure your config is correct and working:
```
Comment (deven98) [inline docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx:102]: Small fix

```suggestion
If you're not sure if you've set up push notifications correctly (e.g. you don't always receive them, they work unreliably), you can follow these steps to make sure your config is correct and working:
```

File: docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx (hunk 2 above)
Comment (imtoori) [reply docusaurus/docs/Flutter/guides/adding_push_notifcations.mdx:188]: lol that's a typo 
```suggestion
As you can see we generate a local notification whenever a message.new or notification.message_new event is received.
```

**Review by deven98 — APPROVED** [review]
LGTM

**Pull request merged**